src/ol/layer/MapboxVector.js

```diff
@@ -92,13 +92,16 @@ export function normalizeStyleUrl(url, token) {
  * Turns mapbox:// source URLs into vector tile URL templates.
  * @param {string} url The source URL.
  * @param {string} token The access token.
+ * @param {string} tokenParam The access token key.
  * @return {string} A vector tile template.
  * @private
  */
-export function normalizeSourceUrl(url, token) {
+export function normalizeSourceUrl(url, token, tokenParam) {
   const mapboxPath = getMapboxPath(url);
   if (!mapboxPath) {
-    return url;
+    return token && url.indexOf(token) === -1
+      ? `${url}${url.indexOf('?') === -1 ? '?' : '&'}${tokenParam}=${token}`
+      : url;
```

Comment:
```suggestion
    if (!token) {
        return url;
    }
    url = new URL(url);
    url.searchParams.set(tokenParam, token);
    return url.href;
```

Maybe we can safely assume that the token doesn't include reserved characters.  But, it feels pretty easy to avoid hand concatenating URLs.

Reply:
Hmm.  Maybe this doesn't cover a case where the source is configured with an `accessToken` but the style URL is not a Mapbox type URL.  Not sure if that is supposed to be handled or not.